src/bhattacharyya.jl

```diff
@@ -50,6 +50,34 @@ end
     return sqab, asum, bsum
 end
 
+@inline function _bhattacharyya_coeff(a::SparseVectorUnion{<:Number}, b::SparseVectorUnion{<:Number})
+    anzind = nonzeroinds(a)
+    bnzind = nonzeroinds(b)
+    anzval = nonzeros(a)
+    bnzval = nonzeros(b)
+    ma = nnz(a)
+    mb = nnz(b)
+
+    ia = 1; ib = 1
+    eval_op = (a, b) -> sqrt(a * b)
+    s = zero(eval_op(zero(eltype(a)), zero(eltype(b))))
+    @inbounds while ia <= ma && ib <= mb
+        ja = anzind[ia]
+        jb = bnzind[ib]
+        if ja == jb
+            v = eval_op(anzval[ia], bnzval[ib])
+            s = v + s
```

Comment:
If `eval_op` is used only once, why not
```suggestion
            s += sqrt(anzval[ia] * bnzval[ib])
```
and remove the function definition above? I guess these are left-overs from your earlier binary_mapreduce1 approach?

Reply:
I made the decision consciously. But, both ways are fine.